tests/Fixtures/Document/TestObject.php

```diff
@@ -0,0 +1,39 @@
+<?php
+/*
+ * Copyright 2023-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace MongoDB\Tests\Fixtures\Document;
+
+final class TestObject
+{
+    public int $id;
+
+    public TestNestedObject $x;
+
+    public bool $decoded;
+
+    public static function createForFixture(int $i, bool $decoded = false): self
+    {
+        $instance = new self();
+        $instance->id = $i;
+        $instance->decoded = $decoded;
```

Comment:
What is the point of this property? Looking at TestDocumentCodec, it looks like the `decode()` method always sets this to `true`. I assume `false` is a default here because you're constructing via a factory method and the resulting object _is not_ the result of `decode()`.

I noted that the use case for calling `createForFixture()` with `$decoded=true` is when asserting decoded documents for an operation. Rather than overload `createForFixture()`, I'd suggest creating a separate factory method that invokes `createForFixture()` and _then_ assigns `$decoded` accordingly. Consider:

```
public static function createDecodedForFixture($int $id): self
{
    $instance = self::createForFixture($id);
    $instance->decoded = true;
    return $instance;
}
```

Then the invocations would be self-documenting.

Along with this, I'd then suggest defaulting the `$decoded` property to `false` in its declaration.

----

I also see that the codec's `encode()` method sets an `$encoded` property on the returned Document instance. Does that pose no problem for PHP 8.2 prohibiting [dynamic property assignment](https://php.watch/versions/8.2/dynamic-properties-deprecated) because we never end up with a TestObject containing an `encoded` property?

Reply:
The `decoded` property allows us to ensure that the object was created through the Codec, and not through other means like a type map (if it ever implements Unserializable down the road).

As for the `encoded` property that's added in `encode()`, that property is never read in `decode`, so it will never get used. That is also only used to check that the object went through the Codec instead of being converted to an object by other means.